### DISCUSSION TITLE
How to preview the SQL code of the query that is about to be executed?

### DISCUSSION THREAD

**Discussion by ppenguin** [ent/ent#1692] [Q&A] [answered]
I'm using `ent` with `postgres`, and for debugging purposes I want to see the SQL code that is executed.

I was surprised there's no such thing like

```golang
...
qb := client.MyData.Query().Where(...).Order(...).Limit(...)

fmt.Sprintf("About to execute query: %s", qb.QueryString())
res, err := qb.All(ctx)
...
```
am I missing something obvious?

**Comment by a8m** [comment, marked as answer]
Hey @ppenguin!

> am I missing something obvious?

Yes. Prefix your queries with `Debug()`. For example:

```go
client.Debug().User.Query().All(ctx)
```

**Comment by YaEvan** [comment]
Another way is to use log level to control the print sql query

`

    drv := entsql.OpenDB(...) / drv, err := entgo.Open(...)
    sqlDrv := dialect.DebugWithContext(drv, func(ctx context.Context, i ...interface{}) {
        // log %v for i
    })
    client := ent.NewClient(ent.Driver(sqlDrv))
`
